### PR TITLE
[MNOE-776] Ability to update_member and remove_member from admin

### PR DESCRIPTION
**Pull request by manu-d**
Needed PUT #update_member and PUT #remove_member actions in admin controller for organizations

=> Move update_member logic in organization model
=> Create shared example for these two methods in a new file

### CODE REVIEW

File: api/app/controllers/mno_enterprise/jpi/v1/admin/organizations_controller.rb [hunk 1]
```diff
@@ -188,6 +211,15 @@ def user_params
       params.require(:user).permit(:email, :name, :surname, :phone)
     end
 
+    def requested_user_in_org(organization)
+      user = organization.users.find { |u| u.id == params.require(:member)[:id] }
```
Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/organizations_controller.rb]: could you please add some comment on why you try first with id and then by email ?

File: core/lib/mno_enterprise/concerns/models/organization.rb [hunk 2]
```diff
@@ -85,6 +85,40 @@ def orga_relation(user)
     }
   end
 
+  # Update a user role within self on behalf on user_updating
+  #
+  # user_updating - A MnoEnterprise::User who performed the action
+  # user_to_update - The user to update. ∈ [MnoEnterprise::User, MnoEnterprise::OrgaInvite]
+  # new_role - The role to assign to user_to_update
+  #
+  def update_user_role(user_updating, user_to_update, new_role)
```
Comment (x4d3) [inline core/lib/mno_enterprise/concerns/models/organization.rb]: I would suggest `user, updated_user`

File: core/lib/mno_enterprise/concerns/models/organization.rb [hunk 3]
```diff
@@ -85,6 +85,40 @@ def orga_relation(user)
     }
   end
 
+  # Update a user role within self
+  #
+  # user - A MnoEnterprise::User who performed the action
```
Comment (ouranos) [inline core/lib/mno_enterprise/concerns/models/organization.rb]: Use [YARD](https://yardoc.org/) format for documentation

eg:
```ruby
# @param user [MnoEnterprise::User] the user performing the action
# @param updated_user [MnoEnterprise::User, MnoEnterprise::OrgaInvite] the user to update
```

File: api/app/controllers/mno_enterprise/jpi/v1/admin/organizations_controller.rb [hunk 4]
```diff
@@ -188,6 +211,17 @@ def user_params
       params.require(:user).permit(:email, :name, :surname, :phone)
     end
 
+    def requested_user_in_org(organization)
+      # If a user has updated their email address, but has not clicked on the confirmation link yet,
+      # we won't find them from their email, so we look from their id first.
+      user = organization.users.find { |u| u.id == params.require(:member)[:id] }
+      user ||= begin
+        email = params.require(:member).require(:email)
+        organization.users.find { |u| u.email == email } ||
```
[the remark this answers is not in the document]
Comment (ouranos) [reply api/app/controllers/mno_enterprise/jpi/v1/admin/organizations_controller.rb]: Also could you pass the entity and id as params this way you can simplify the controller logic a bit?

**Pull request merged**